sgx-hasher/src/lib.rs

```
@@ -79,6 +80,52 @@ impl Hasher {
     }
 }
 
+pub fn sign(
+    author: sig::Author,
+    contents: sig::Contents,
+    key: rsa::Rsa<pkey::Private>,
+) -> std::io::Result<sig::Signature> {
+    // Generates signature on Signature author and contents
+    let rsa_key = pkey::PKey::from_rsa(key.clone())?;
+    let md = hash::MessageDigest::sha256();
+    let mut signer = sign::Signer::new(md, &rsa_key)?;
+    signer.update(author.as_ref())?;
+    signer.update(contents.as_ref())?;
+    let signature = signer.sign_to_vec()?;
+
+    // Generates q1, q2 values for RSA signature verification
+    let s = bn::BigNum::from_slice(&signature)?;
+    let m = key.n();
+    let e = u32::try_from(key.e()).expect("Could not convert BigNum to u32");
+    let mut ctx = bn::BigNumContext::new()?;
+    let mut q1 = bn::BigNum::new()?;
+    let mut qr = bn::BigNum::new()?;
+
+    q1.div_rem(&mut qr, &(&s * &s), &m, &mut ctx)?;
+    let q2 = &(&s * &qr) / m;
+
+    // Returns modulus, signature, q1, and q2 as [u8; 384]
+    let mut m_array = [0; 384];
+    m.to_slice(&mut m_array).unwrap_or(());
```

`.unwrap_or(())` just swallows the error. Silent failures aren't good.